driver/src/main/java/org/neo4j/driver/internal/messaging/common/CommonValueUnpacker.java

```diff
@@ -374,34 +377,26 @@ private Value unpackLocalDateTime() throws IOException {
         return value(LocalDateTime.ofEpochSecond(epochSecondUtc, nano, UTC));
     }
 
-    private Value unpackDateTimeWithZoneOffset() throws IOException {
-        long epochSecondLocal = unpacker.unpackLong();
-        int nano = Math.toIntExact(unpacker.unpackLong());
-        int offsetSeconds = Math.toIntExact(unpacker.unpackLong());
-        return value(newZonedDateTime(epochSecondLocal, nano, ZoneOffset.ofTotalSeconds(offsetSeconds)));
-    }
-
-    private Value unpackDateTimeUtcWithZoneOffset() throws IOException {
-        long epochSecondLocal = unpacker.unpackLong();
-        int nano = Math.toIntExact(unpacker.unpackLong());
-        int offsetSeconds = Math.toIntExact(unpacker.unpackLong());
-        ZoneOffset offset = ZoneOffset.ofTotalSeconds(offsetSeconds);
-        return value(newZonedDateTimeUsingUtcBaseline(epochSecondLocal, nano, offset));
-    }
-
-    private Value unpackDateTimeWithZoneId() throws IOException {
-        long epochSecondLocal = unpacker.unpackLong();
-        int nano = Math.toIntExact(unpacker.unpackLong());
-        String zoneIdString = unpacker.unpackString();
-        return value(newZonedDateTime(epochSecondLocal, nano, ZoneId.of(zoneIdString)));
-    }
-
-    private Value unpackDateTimeUtcWithZoneId() throws IOException {
-        long epochSecondLocal = unpacker.unpackLong();
-        int nano = Math.toIntExact(unpacker.unpackLong());
-        String zoneIdString = unpacker.unpackString();
-        ZoneId zoneId = ZoneId.of(zoneIdString);
-        return value(newZonedDateTimeUsingUtcBaseline(epochSecondLocal, nano, zoneId));
+    private Value unpackDateTime(boolean unpackOffset, boolean useUtcBaseline) throws IOException {
```

Comment:
nitpick: the usage of this method like `unpackDateTime(false, true)` doesn't give much context about the expected behaviour.

Reply:
This is just an internal small method with 2 arguments, but I have added enums for readability.